controllers/mod/search.rb

```diff
@@ -0,0 +1,71 @@
+class ArtefactsMetadataController < ApplicationController
+  namespace "/mod-api" do
+    namespace "/search" do
+      
+      doc('Search', 'Search content/metadata of artefacts') do
+        parameter('q', type: 'string', description: 'Query text', default: 'plant')
+        parameter('page', type: 'integer', description: 'Page number', default: '1')
+        parameter('pagesize', type: 'integer', description: 'Number of items per page', default: '10')
+        parameter('display', type: 'string', description: 'Attributes to display', default: '')
+        response(200, "OK", content('$ref' => '#/components/schemas/hydraPage'))
+      end
+      get do
+        result = process_search
+        reply hydra_page_object(result.to_a, result.aggregate)
+      end
+
+      doc('Search', 'Search content of artefacts') do
+        parameter('q', type: 'string', description: 'Query text', default: 'plant')
+        parameter('page', type: 'integer', description: 'Page number', default: '1')
+        parameter('pagesize', type: 'integer', description: 'Number of items per page', default: '10')
+        parameter('display', type: 'string', description: 'Attributes to display', default: '')
+        response(200, "OK", content('$ref' => '#/components/schemas/hydraPage'))
+      end
+      get '/content' do
+        result = process_search
+        reply hydra_page_object(result.to_a, result.aggregate)
+      end
+
+      doc('Search', 'Search metadata of artefacts') do
+        parameter('q', type: 'string', description: 'Query text', default: 'STY')
+        parameter('page', type: 'integer', description: 'Page number', default: '1')
+        parameter('pagesize', type: 'integer', description: 'Number of items per page', default: '10')
+        parameter('display', type: 'string', description: 'Attributes to display', default: '')
+        response(200, "OK", content('$ref' => '#/components/schemas/hydraPage'))
+      end
+      get '/metadata' do
+        query = get_query(params)
```

Comment:
extract the search logic into a helper as done with  process_search, to leave this controller short and clean